docs/design/proposals/VRG-Reconciliation.md

```diff
@@ -0,0 +1,150 @@
+# VRG reconciler workflow
+
+This design document covers the VRG CRD and its reconciliation workflow.
+
+**NOTE**: This is added to aid understanding and ease development of the
+various parts in parallel (where possible) and would possibly be merged once
+it reaches initial completion.
+
+## Terminology
+
+- VRG: VolumeReplicationGroup
+- VR: VolumeReplication
+- PVC: PersistentVolumeClaim
+- PV: PersistentVolume
+
+## VRG CRD
+
+- Spec:
+    - PVCSelector
+        - Label selector to identify all the PVCs that are in this group that
+        needs to be replicated to the peer cluster.
+    - VolumeReplicationClass
+        - ReplicationClass of all volumes in this replication group; this
+        value is propagated to children VolumeReplication CRs
+    - ReplicationState
+        - Desired state of all volumes [`primary` or `secondary`] in this
+        replication group; this value is propagated to children
+        VolumeReplication CRs
+    - S3Endpoint
+        - S3 Endpoint to replicate PV metadata
+    - S3SecretName
+        - S3 Endpoint to replicate PV metadata
+- Status:
```

Comment:
Need to add S3Region.  But, this can also wait till we resolve the S3 secret design (another thread).